### PR TITLE
feat: integration tests framework

### PR DESCRIPTION
**Pull request by rpiszczatowski**

### CODE REVIEW

File: packages/cache-service/test/e2e/data-packages.spec.ts [hunk 1]
```diff
@@ -363,7 +363,7 @@ describe("Data packages (e2e)", () => {
     const parsedDataPoints = JSON.parse(allFeedsDataPackages[0]).dataPoints;
     expect(allFeedsDataPackages.length).toBe(4);
     expect(parsedDataPoints.length).toBe(3);
-    for (const [_, dataPackages] of Object.entries<any>(testResponse2.body)) {
+    for (const [, dataPackages] of Object.entries<any>(testResponse2.body)) {
```
Comment (cehali) [inline packages/cache-service/test/e2e/data-packages.spec.ts:366]: Can we change `any` to data packages type?

File: packages/integration/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
```
Comment (cehali) [inline packages/integration/.eslintrc.js:1]: Have you checked if it is compatible with our prettier settings? We don't want to have conflicts between these frameworks.

File: packages/integration/integration-test-classic.ts [hunk 3]
```diff
@@ -94,14 +58,39 @@ const main = async () => {
...
       ],
       "relayer-contract",
+      { ADAPTER_CONTRACT_ADDRESS: adapterContractAddress },
       false
     );
+
+const main = async () => {
+  cacheLayerInstance = await startAndWaitForCacheLayer("1");
+  oracleNodeInstance = await startAndWaitForOracleNode(
+    "1",
```
Comment (cehali) [inline packages/integration/integration-test-classic.ts:68]: Please extract `1` to separate variable called `instanceId` or something similar

File: packages/integration/integration-test-classic.ts [hunk 4]
```diff
@@ -94,14 +58,39 @@ const main = async () => {
...
+  );
 
   process.exit();
 };
 
-common.configureCleanup(stopAll);
+configureCleanup(stopAll);
 
-main();
+void main();
```
Comment (cehali) [inline packages/integration/integration-test-classic.ts:96]: Do we need `void` here?

File: packages/integration/integration-test-core.ts [hunk 5]
```diff
@@ -1,22 +1,46 @@
...
 
   process.exit();
-}
+};
 
-common.configureCleanup(stopAll);
+configureCleanup(stopAll);
 
-main();
+void main();
```
Comment (cehali) [inline packages/integration/integration-test-core.ts:46]: Do we need void here?

**Review by cehali — DISMISSED** [review]
Good job, small things in comments

File: packages/integration/.eslintrc.js (hunk 2 above)
Comment (rpiszczatowski) [reply packages/integration/.eslintrc.js:1]: I checked that but I'll additionally add eslint-prettier-plugin.

File: packages/integration/integration-test-classic.ts (hunk 4 above)
Comment (rpiszczatowski) [reply packages/integration/integration-test-classic.ts:96]: This is a price we pay for having "@typescript-eslint/no-floating-promises" eslint rule enabled. The void cast disables this check and here we simply cannot use await keyword. I'll change this to explicitly disabling the rule for this line.

File: packages/cache-service/test/e2e/data-packages.spec.ts (hunk 1 above)
Comment (rpiszczatowski) [reply packages/cache-service/test/e2e/data-packages.spec.ts:366]: My understanding of the code below is that `dataPackages` variable type is `string[]` as we access it as an array and pass this array's elements to `JSON.parse`. I'll then leave it as is.

**Pull request merged**